### PR TITLE
Yaldi gui home

### PR DESCRIPTION
**Pull request by azkan01**
1. Added package.json for gulp-concat to assist with minification.

2. Gulp task added to minify - remaining questions around implementing this. Task not is use but it available for when required. 

3. Added new search element as the main focal point of home page and associated styling. Controller has two main functions. To open the datepicker and execute a dumb search - next step for this would be implementing an api call for search. Tests included.

4. Result element added to display currently hardcoded search results. As above, this will be enriched when the api call is added. 

5. Home layout stripped down.

6. Navbar elements removed to simplify navigation - stripped to base layer. enrichment to follow. 

7. Font files added - POC for adding and using fonts. Not sure on the styling - group decision will need to be reached but will be handled as future story.

### CODE REVIEW

File: src/main/webapp/app/components/login/login.html [hunk 1]
```diff
@@ -1,19 +1,20 @@
...
+        <!--div class="col-md-4 col-md-offset-4">
             <h1 data-translate="login.title">Sign in</h1>
         </div>
         <div class="col-md-8 col-md-offset-2">
             <div class="alert alert-danger" ng-show="vm.authenticationError" data-translate="login.messages.error.authentication">
                 <strong>Failed to sign in!</strong> Please check your credentials and try again.
             </div>
-        </div>
-        <div class="col-md-6">
+        </div-->
```
Comment (chriskiernan87) [inline src/main/webapp/app/components/login/login.html]: Is this supposed to be commented? If so just remove the code.

File: src/main/webapp/app/components/login/login.html [hunk 2]
```diff
@@ -33,19 +34,26 @@ <h1 data-translate="login.title">Sign in</h1>
...
+            <div class="alert alert-danger" ng-show="vm.authenticationError" data-translate="login.messages.error.authentication">
+                <strong>Failed to sign in!</strong> Please check your credentials and try again.
+            </div>
         </div>
-        <div class="col-md-6">
+        <!--div class="col-md-6">
             <br/>
             <jh-social ng-provider="google"></jh-social>
             <jh-social ng-provider="facebook"></jh-social>
             <jh-social ng-provider="twitter"></jh-social>
```
Comment (chriskiernan87) [inline src/main/webapp/app/components/login/login.html:48]: Commented code.

File: src/main/webapp/app/home/home.html [hunk 3]
```diff
@@ -1,35 +1,8 @@
 <div ng-cloak>
    <div class="row">
+       	<div class="col-md-6 col-md-offset-3">
+           <!--div ng-switch="vm.isAuthenticated()"></div-->
```
Comment (chriskiernan87) [inline src/main/webapp/app/home/home.html]: commented code

File: src/main/webapp/app/layouts/footer/footer.html [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        <li><a ui-sref="home">News</a></li>
+      </ul>
+    </div>
+    <div class="footer-right">
+      <ul style="list-style-type:none">
+        <li><a ui-sref="home">Useful Links</a></li>
+      </ul>
+    </div>
+  </div>
+  <!--div class="col-md-2"> </div-->
```
Comment (chriskiernan87) [inline src/main/webapp/app/layouts/footer/footer.html]: commented code

File: src/main/webapp/app/layouts/navbar/navbar.html [hunk 5]
```diff
@@ -1,31 +1,41 @@
-<nav class="navbar navbar-default" role="navigation">
+<nav class="navbar navbar-default " role="navigation">
     <div class="container">
         <div class="navbar-header">
-            <button type="button" class="navbar-toggle" ng-click="vm.toggleNavbar()">
+            <!--button type="button" class="navbar-toggle" ng-click="vm.toggleNavbar()">
                 <span class="sr-only">Toggle navigation</span>
                 <span class="icon-bar"></span>
                 <span class="icon-bar"></span>
```
Comment (chriskiernan87) [inline src/main/webapp/app/layouts/navbar/navbar.html]: commented code

File: src/main/webapp/app/layouts/navbar/navbar.html [hunk 6]
```diff
@@ -129,20 +127,20 @@
...
+                <!--li ui-sref-active="active" uib-dropdown class="dropdown pointer" ng-controller="JhiLanguageController as languageVm">
                     <a class="dropdown-toggle" uib-dropdown-toggle href="" ng-if="languageVm.languages.length > 1">
                         <span>
                             <span class="hidden-sm" data-translate="global.menu.language">Language</span>
-                            <b class="caret"></b>
                         </span>
                     </a>
                     <ul class="dropdown-menu" uib-dropdown-menu ng-if="languageVm.languages.length > 1">
                         <li active-menu="{{language}}" ng-repeat="language in languageVm.languages">
                             <a href="" ng-click="languageVm.changeLanguage(language);vm.collapseNavbar();">{{language | findLanguageFromKey}}</a>
```
Comment (chriskiernan87) [inline src/main/webapp/app/layouts/navbar/navbar.html]: commented code

File: src/main/webapp/app/layouts/search/results.controller.js [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+						 "address" : "ADD5",
+						 "rating" : "5",
+						 "url" : "" },
+						{
+						 "name" : "No6", 
+						 "city" : "City6",
+						 "postcode" : "PC6",
+						 "address" : "ADD6",
+						 "rating" : "1",
+						 "url" : "" },
```
Comment (chriskiernan87) [inline src/main/webapp/app/layouts/search/results.controller.js:58]: what's with the hard coded values?

File: src/main/webapp/app/layouts/search/results.html [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+</div>
+
+<div class="container-fluid">
+	<!--div class="col-md-2"> </div-->
+	<div class="col-md-8 col-md-offset-2"> 
+			<div class="card" ng-repeat="user in vm.users">
+			  <img class="card-img-left" src="../../content/images/logo-jhipster.png" alt="Card image cap">
+			  <div class="card-block">
+			    <h4 class="card-title">{{user.name}}</h4>
+			    <p class="card-text"> Description Description Description Description Description Description Description Description </p>
```
Comment (chriskiernan87) [inline src/main/webapp/app/layouts/search/results.html:11]: description?

File: src/main/webapp/app/layouts/search/results.html [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+			    <div class="row" align="centre">
+			    	rating
+		            <i class="fa fa-star" ng-if="user.rating >= 1"></i>	    
+		            <i class="fa fa-star" ng-if="user.rating >= 2"></i>	    	
+		            <i class="fa fa-star" ng-if="user.rating >= 3"></i>	    	
+		            <i class="fa fa-star" ng-if="user.rating >= 4"></i>	    	
+		            <i class="fa fa-star" ng-if="user.rating >= 5"></i>	    		
+			    </div>
+			    <!--a href="#" class="btn btn-primary inline">Visit Page</a>
+			    <a href="#" class="btn btn-primary inline">Book Now</a>
```
Comment (chriskiernan87) [inline src/main/webapp/app/layouts/search/results.html]: commented code

File: src/main/webapp/app/layouts/search/search.html [hunk 10]
```diff
@@ -0,0 +1,26 @@
+<div class="row">
+	<!--div class="col-md-3"/-->
+	<div class="col-md-6 col-md-offset-3"> <h1 class="tagline"> Yalday! Where to next? </h1> </div>
+	<!--div class="col-md-3"/-->
```
Comment (chriskiernan87) [inline src/main/webapp/app/layouts/search/search.html]: commented code

File: src/main/webapp/index.html [hunk 11]
```diff
@@ -16,6 +16,9 @@
     <!-- endbuild -->
     <!-- build:css content/css/main.css -->
     <link rel="stylesheet" href="content/css/main.css">
+    <link href="//maxcdn.bootstrapcdn.com/font-awesome/4.1.0/css/font-awesome.min.css" rel="stylesheet">
+    <script src="https://use.fontawesome.com/958b12b81f.js"></script>
+    <!--link rel="stylesheet" href="http://ajax.googleapis.com/ajax/libs/angular_material/1.1.0/angular-material.min.css"-->
```
Comment (chriskiernan87) [inline src/main/webapp/index.html]: commented code

File: src/test/javascript/spec/app/layouts/navbar/navbar.controller.spec.js [hunk 12]
```diff
@@ -0,0 +1,8 @@
+'use strict';
+
+describe('Controller Tests', function() {
+
```
Comment (chriskiernan87) [inline src/test/javascript/spec/app/layouts/navbar/navbar.controller.spec.js]: test file with no tests?

File: src/test/javascript/spec/app/layouts/search/results.controller.spec.js [hunk 13]
```diff
@@ -0,0 +1,8 @@
+'use strict';
+
+describe('Controller Tests', function() {
+
+    describe('Results Controller Tests', function() {
```
Comment (chriskiernan87) [inline src/test/javascript/spec/app/layouts/search/results.controller.spec.js]: test file with no tests

File: src/main/webapp/app/layouts/search/results.controller.js (hunk 7 above)
Comment (azkan01) [reply src/main/webapp/app/layouts/search/results.controller.js:58]: Just test data for displaying search results. Will be amended when we plug in the endpoint call for searching the database. 

I can remove it if you like.

File: src/main/webapp/app/layouts/search/results.html (hunk 8 above)
Comment (azkan01) [reply src/main/webapp/app/layouts/search/results.html:11]: Test data to check how it resolves on the front end. Again, would be removed when the endpoint call is implemented - probably the next story for me. 

Shall I remove?

File: src/test/javascript/spec/app/layouts/search/results.controller.spec.js (hunk 13 above)
Comment (azkan01) [reply src/test/javascript/spec/app/layouts/search/results.controller.spec.js]: no functions to test. Removing for now - again, will be required when the endpoint used.

**Comment by azkan01** [comment]
Comments actioned - ready for re-review when you have time.

**Pull request merged**